datalad/interface/utils.py

```diff
@@ -107,6 +107,33 @@ def handle_dirty_dataset(ds, mode, msg=None):
         raise ValueError("unknown if-dirty mode '{}'".format(mode))
 
 
+def common_root_ds(paths):
+    """Return a common root dataset for a set of absolute paths
+
+    Note, that in opposition to get_tree_roots, this will figure a *common* root, not a set of roots.
+    In addition, this currently does actually resolve the paths rather than performing pure path operations.
+    """
+
+    from collections import Counter
+    from datalad.utils import assure_list
+    from datalad.distribution.dataset import get_dataset_root
+
+    paths = assure_list(paths)
+    paths = [p.resolve() if p.exists() else p for p in paths]
+    counter = Counter()
+    # Note, we need the stringified paths for sorting as well as to pass the common_parent into get_dataset_root
+    counter.update([str(p) for p in paths])
+    for p in paths:
+        [counter.update(str(par)) for par in p.parents]
```

Comment:
this might end up quite inefficient, would be `len(paths) * len(parents)`  but what is needed to just "shrink down" common prefix, pretty much taking parents of the first path, then retaining the matching tail of that list with the next path, and break early if list gets empty (no common_parent), or then taking the first (longest) element

Reply:
and [get_parents](https://github.com/datalad/datalad/blob/master/datalad/support/path.py#L119) sounds like something we might be able to use it here? e.g. given paths with parents of the first path, and some post-analysis.... not yet sure, didn't try but i feel it was solving a similar problem and avoided `len(paths) * len(parents)` runtime

Reply:
Re 1: Yep. Thought of that, too.

Re 2: Thanks for pointing to it. Will check. And even if I can't make use of it, that common parent detection should probably move there (from current interface/utils.py)

Reply:
However, what's probably more important re performance is, whether or not to resolve those paths in order to find a common dataset they belong to. If we don't, we'll miss out on some valid cases not recognizing they belong to the same thing. But then - to be completely correct we would actually need to discover actual repositories bottom up to account for all the fancy things you can do with symlinks + `.git`, etc.
That'd be really expensive though.

Not resolving and do plain path operation not checking FS at all would be fast and still cover a lot of cases, I think. Might be acceptable.